src/parser.ts

```diff
@@ -13,6 +13,21 @@ let PlaceHolders = {
     ArrowParameterPlaceHolder: 'ArrowParameterPlaceHolder'
 };
 
+function upgradeToken(token) {
```

Comment:
That would defeat the point. A scanner should never need to know whether the current parsing happens in the context of strict mode or not. This "upgrade" mechanism is an artifact, a historical baggage that would have not exist if the design was different.